src/commonMain/kotlin/com/github/nanoflakes/NanoflakeWIthEpochGenerator.kt

```diff
@@ -0,0 +1,8 @@
+package com.github.nanoflakes
+
+/**
+ * A generator of [nanoflakes][Nanoflake] that retains its epoch information.
+ */
+public interface NanoflakeWithEpochGenerator : NanoflakeGenerator {
```

Comment:
I actually strongly disagree with the idea of an `NanoflakeWithEpochGenerator`

I'd prefer that this interface gets deleted, and the Nanoflake interface gets a "hasEpoch" boolean getter and "withEpoch" extension functions.

Reply:
I meant to bring this up as I wasn't too sure how I felt about it either, guess I forgot to over the week.
>  the Nanoflake interface gets a "hasEpoch" boolean...

Do you mean the `NanoflakeGenerator`?

> "withEpoch" extension functions.

How would this work? The "main" function of a NanoflakeGenerator is the `next` operator function. Are you suggesting to add a `nextWithEpoch` function? In that case users cannot benefit from `next` being an operator function.